### ISSUE TITLE
no linting is available in the project

### ISSUE THREAD

**Issue by nils-a** [cake-build/cake-vscode#475]
The project has no linting, currently 
#464 removed the reference to `tslint` (though it is not mentioned in the commit)

While it might have been better to add a lint script in `package.json` instead of removing `tsling`, the [recommendation is `eslint`, now](https://code.visualstudio.com/api/advanced-topics/tslint-eslint-migration).

Having a linter would be a good start towards a common style (and also to avoid common mistakes).

**Comment by pascalberger** [comment]
There would also be a ESLint issue provider for Cake.Issues if you want to have it included in the build process (easiest way probably by adding Cake.Issues.Recipe to Cake.VsCode.Recipe, like for Cake.Recipe)

**Comment by nils-a** [comment]
@pascalberger that sounds like a good idea. Additionally we should add the linting, as well as running the tests (and probably depcheck, too) to the ci build. Sounds like a new issue. @gep13 What do you think?

**Comment by pascalberger** [comment]
I created https://github.com/cake-contrib/Cake.VsCode.Recipe/issues/12 for addin Cake.Issues.Recipe to Cake.VsCode.Recipe. 

The question is who is responsible for running the linter. I saw you added it as a node script with https://github.com/nils-a/cake-vscode/commit/5cdfdd106a3c0790f34e88aa2a006c44897deabf#. From a recipe point of view it would be cleaner if this is all done by Cake.VsCode.Recipe so that it doesn't require specific setup in the project which it builds.

**Comment by gep13** [comment]
@pascalberger agreed, if we are going to add this in (behind a toggle so that it can be switched off) it would make sense to go into Cake.VsCode.Recipe, so that other extensions using the Recipe can take advantage of it.  Seems like we are already going to have to get a release of Cake.VsCode.Recipe out the door, so we could sneak this in before shipping that.

**Comment by nils-a** [comment]
@pascalberger yes - I did that with yesterday's "this is a node project"-view of the world 😄 

From a Recipe-view we would best create `Cake.ESLint`. I guess the thing I'm trying to say is that, having eslint installed creates scripts `node_modules/.bin/eslint[|.cmd|.ps1]` and we would need to decide which one to call. using `npm run lint` is easy and does that for us - provided the npm script `lint` is available.

On the other hand - eslint commandline options seem not too overwhelming so a first usable version is probably rather quickly written down.

**Comment by gep13** [comment]
Seems like there is an open issue about this already: https://github.com/cake-contrib/addin-requests/issues/6

**Comment by pascalberger** [comment]
You don't necessarily need a plugin to run ESLint (or any npm commend). It's also perfectly fine to just run the process directly using any of Cakes [process aliases](https://cakebuild.net/dsl/process/). The question should be "Who is in control of the build orchestration" ? Is it Cake? Is it npm scripts? Having one tool responsible for orchestration calling other tools, is perfectly fine. As soon as there are different orchestrators it starts to become complicated.

**Comment by gep13** [comment]
Given that there is more to the overall build process, I personally would prefer that Cake is doing the work, rather than going into npm scripts doing the work, but that is just me.

**Comment by nils-a** [comment]
Agreed. I'll see if I can switch from `package.json` scripts to cake.

**Comment by nils-a** [comment]
@gep13 @pascalberger What do you think of https://github.com/nils-a/cake-vscode/commit/cd93fb7e89a8133a3a91bcff6496eb9fc1a01025 ?

**Comment by pascalberger** [comment]
> @gep13 @pascalberger What do you think of [nils-a@cd93fb7](https://github.com/nils-a/cake-vscode/commit/cd93fb7e89a8133a3a91bcff6496eb9fc1a01025) ?

The question is if running linters and tests is something which we expect from the recipe script or is it something which every build script needs to implement itself? At a first glance I would say this is something which should be possible to add to Cake.VsCode.Recipe.

**Comment by gep13** [comment]
@nils-a as a first pass, I think the tasks make a lot of sense, but I also agree with @pascalberger it would be good to get these added into the Recipe, so that other users don't also have to implement these steps, but rather simply have a parameter they can put in their recipe.cake file to turn it off if they don't want it.

**Comment by nils-a** [comment]
Related to (and partially duplicates) #78
